### ISSUE TITLE
`Generic[_T]` lost on `DataUpdateCoordinator`

### ISSUE THREAD

**Issue by scop** [KapJI/homeassistant-stubs#290]
https://github.com/home-assistant/core/blob/3d8e8293d67094da383bb57f518802784e1fe3a5/homeassistant/helpers/update_coordinator.py#L55 vs
https://github.com/KapJI/homeassistant-stubs/blob/98df454159346546581bd0cd540c1945870d46cb/homeassistant-stubs/helpers/update_coordinator.pyi#L26 -- `Generic[_T]` got lost.

I suppose this is why for something like https://github.com/scop/home-assistant-jatekukko/blob/66ceb35160f20026d3f638e7696513a585a5436f/custom_components/jatekukko/coordinator.py#L42, mypy gives:
```
custom_components/jatekukko/coordinator.py:42: error: "DataUpdateCoordinator" expects no type arguments, but 1 given  [type-arg]
```

Similarly for `BaseCoordinatorEntity`.

This is with homeassistant-stubs 2023.2.4.

**Comment by KapJI** [comment]
Unfortunately, this is limitation of `stubgen`, it doesn't extract all info, especially in more complicated cases like this. I really think that HA should just export typing info directly even if it's not fully complete.

**Comment by KapJI** [comment]
Should be solved by mypy 1.4 upgrade in homeassistant-stubs 2023.6.3.
